python_modules/dagster/dagster/_core/captured_log_api.py

```diff
@@ -0,0 +1,36 @@
+import base64
+from typing import NamedTuple, Sequence
+
+from dagster._seven import json
+
+# This should be emitted as a log message to indicate that no more logs will be emitted for that process
+# this lets the log reader know that it can stop polling for more logs
+LOG_STREAM_COMPLETED_SIGIL = "LOGS COMPLETED"
```

Comment:
having to remember to emit this log whenever a process completes so that we know the logs are over seems brittle...

Reply:
do we have a backfill completed / backfill failed log message? maybe we should just key on that?

or come up with some other scheme ie just see the backfill is in a terminal state and expect no more things to be added 

the latter would mean probably two layer interpretation, where the compute log methods return whether or not it knows there are are more and then at the layer above you combine that information with knowing if the thing producing logs is done to determine the final hasMore

Reply:
there is a log message at the end of the backfill we could look for. I was thinking about if we wanted to use this for non-backfill logs we'd want to have some kind of shared indicator that the logs are done

Reply:
ended up changing the meaning of `has_more` for the captured log manager. It returns true only if there are more logs it could read right then. Then the caller can compare that `has_more` value with what it knows about the state of the process to determine if there are actually more logs. So for our case, if the log manager says `has_more=False` and the backfill is in the COMPLETED state then we know there are no more logs

Reply:
maybe rename to `has_more_now` in the lower layer for clarity